js/modules/k6/k6.go

```diff
@@ -22,6 +23,9 @@ var (
 	ErrCheckInInitContext = common.NewInitContextError("Using check() in the init context is not supported")
 )
 
+const asyncFunctionNotSupportedMsg = "%s: async functions are not supported as arguments, " +
+	"please see https://github.com/grafana/k6/issues/2728#issuecomment-1404747660 for more info"
```

Comment:
```suggestion
const asyncFunctionNotSupportedMsg = "async functions are currently unsupported as %s() arguments"
```

I don't think we need to link to an issue comment in an exception message, that would just add noise. And if we really think we should attach a longer explanation, then we should link to a page in the docs or a standalone issue with the information, https://github.com/grafana/k6/issues/2728 has way too many comments and unrelated discussion.

Reply:
> , that would just add noise

Given that this error means your code doesn't work at all - adding noise seems like a good idea. I even wondered if I shouldn't abort the whole test but decided against that based on complexity. 

> should link to a page in the docs or a standalone issue with the information

That is probably better, but likely will not be any less noisy  and will mean that we will need to make a page specifically for that.

Linking the specific comment IMO is best -if somebody wants to use async function the specifically linked comment goes over why this is currently not supported and the issue has more information if somebody needs it.

If this is a problem arguably us being told that users are hitting it is a good idea.

Reply:
I am not convinced :thinking: 

> Given that this error means your code doesn't work at all

How is that any different than any other error? :sweat_smile: Though as I wrote in https://github.com/grafana/k6/issues/1681, I do want identifiable and easily searchable error IDs, so maybe we can start here :thinking: How about
```suggestion
const asyncFunctionNotSupportedMsg = "async functions are currently unsupported as %s() arguments " +
	" (k6AsyncCallbacksForbidden)";
```

> Linking the specific comment IMO is best -if somebody wants to use async function the specifically linked comment goes over why this is currently not supported and the issue has more information if somebody needs it.

I would be fine with providing more information in the error, if it contained some useful workaround for users. But sending them on a quest to read a long issue, only to indeed discover that there is no way to use `async` and `group`, that seems a bit rude :sweat_smile: 

Even if we ignore UX and the main goal is to collect more feedback, this is probably not the right way to go about it and opening a new issue with that summary a link to the old ones is better :thinking:

Reply:
> Even if we ignore UX and the main goal is to collect more feedback, this is probably not the right way to go about it and opening a new issue with that summary a link to the old ones is better

A humble suggestion to update the doc for `group()` describing the reason for this limitation there and reference it in the error message.

Reply:
>Even if we ignore UX and the main goal is to collect more feedback,

I think it is important in this case, because the API is under discussion and maybe a user that hits this path has a  use case that we would be good to know. If we can have a cleaner issue I'm happy with it, but I think the current is better than nothing.

```suggestion
const asyncFunctionNotSupportedMsg = "%s: async functions are not supported as arguments, " +
	"check https://github.com/grafana/k6/issues/2728#issuecomment-1404747660 for sharing your experience and read about more details"
```

Reply:
WDYT https://github.com/grafana/k6-docs/pull/1011

Reply:
WDYT about the new URL. We can make the same message for the `check` API as well and redirect to the correct one.

Or leave it like that - I doubt anyone will try to use `check` with async functions :crossed_fingers:

Reply:
I think it's fine to leave it as it is, I very much doubt that someone will ever use an `async` function in a `check()`, but if they do, linking to the group warning is good enough